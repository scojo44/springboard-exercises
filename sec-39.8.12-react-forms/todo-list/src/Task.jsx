import React from 'react'

function Task({id, task, remove, update }) {
  /** Have TodoList know remove the todo from the list */
  function removeClicked(e) {
    remove(id);
  }

  /** Let TodoList know this todo is being edited */
  function editClicked(e) {
    update(id, task, true);
  }

  return (
    <>
      <div>{task}</div>
      <div><button onClick={editClicked}>Edit</button></div>
      <div><button onClick={removeClicked}>X</button></div>
    </>
  )
}

export default Task
