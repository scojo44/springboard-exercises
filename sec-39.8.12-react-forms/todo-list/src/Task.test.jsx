import { render, fireEvent } from '@testing-library/react';
import Task from './Task';

describe('Task Tests', () => {
  it('renders without crashing', () => {
    render(<Task id={1} task="Learn Docker" />);
  });

  it('matches snapshot', () => {
    const {asFragment} = render(<Task id={1} task="Learn Docker" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('shows a task with Edit and Remove buttons', () => {
    const {getByText} = render(<Task id={1} task="Learn Docker" remove={x => x} update={x => x} />);
    const task = getByText('Learn Docker');
    const editButton = getByText('Edit');
    const removeButton = getByText('X');

    expect(task).toBeInTheDocument();
    expect(editButton).toBeInTheDocument();
    expect(removeButton).toBeInTheDocument();
  });

  it('should call the passed remove function when X is clicked', () => {
    const remove = vitest.fn();
    const {getByText} = render(<Task id={1} task="Learn Docker" remove={remove} />);
    const button = getByText('X');

    expect(button).toBeInTheDocument();
    fireEvent.click(button);
    expect(remove).toHaveBeenCalledWith(1);
  });

  it('should call the passed update function when Edit is clicked', () => {
    const update = vitest.fn();
    const {getByText} = render(<Task id={1} task="Learn Docker" update={update} />);
    const button = getByText('Edit');

    expect(button).toBeInTheDocument();
    fireEvent.click(button);
    expect(update).toHaveBeenCalledWith(1, "Learn Docker", true);
  });
});
