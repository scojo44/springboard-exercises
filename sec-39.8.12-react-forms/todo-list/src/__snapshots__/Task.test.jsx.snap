// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Task Tests > matches snapshot 1`] = `
<DocumentFragment>
  <div>
    Learn Docker
  </div>
  <div>
    <button>
      Edit
    </button>
  </div>
  <div>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
